src/plugins/intel_cpu/src/cpu_streams_calculation.cpp

```diff
@@ -40,43 +41,53 @@ std::vector<std::vector<int>> get_streams_info_table(const int input_streams,
-        int numa_node_id = stream_info[STREAM_NUMA_NODE_ID];
         int socket_id = stream_info[STREAM_SOCKET_ID];
         int node_start = one_proc_table.size() == 1 ? 0 : 1;
         int node_end = one_proc_table.size() == 1 ? 1 : one_proc_table.size();
-        for (int n = MAIN_CORE_PROC; n <= HYPER_THREADING_PROC; n++) {
-            for (int index = node_start; index < node_end; index++) {
-                if (((numa_node_id < 0) || (numa_node_id == one_proc_table[index][PROC_NUMA_NODE_ID])) &&
-                    ((socket_id < 0) || (socket_id == one_proc_table[index][PROC_SOCKET_ID]))) {
-                    if (0 != one_proc_table[index][n]) {
-                        stream_info[PROC_TYPE] = n;
-                        if (total_threads <= one_proc_table[index][n]) {
-                            stream_info[THREADS_PER_STREAM] = total_threads;
-                            stream_info[STREAM_NUMA_NODE_ID] = one_proc_table[index][PROC_NUMA_NODE_ID];
-                            stream_info[STREAM_SOCKET_ID] = one_proc_table[index][PROC_SOCKET_ID];
-                            streams_info_table.push_back(stream_info);
-                            return;
-                        } else {
-                            stream_info[THREADS_PER_STREAM] = one_proc_table[index][n];
-                            stream_info[STREAM_NUMA_NODE_ID] = one_proc_table[index][PROC_NUMA_NODE_ID];
-                            stream_info[STREAM_SOCKET_ID] = one_proc_table[index][PROC_SOCKET_ID];
-                            streams_info_table.push_back(stream_info);
-                            total_threads -= one_proc_table[index][n];
+        // When n_mode is 3, the following loop only selects CPUs on socket with the same id as current_socket_id.
+        // When n_mode is 2, the following loop only selects CPUs on sockets with id different from current_socket_id.
+        // When n_mode is 1, the following loop selects CPUs on all sockets.
+        for (int n_mode = current_socket_id < 0 ? 1 : 3; (n_mode > 0) && (total_threads > 0); n_mode--) {
+            for (int n = MAIN_CORE_PROC; (n <= HYPER_THREADING_PROC) && (total_threads > 0); n++) {
+                for (int index = node_start; (index < node_end) && (total_threads > 0); index++) {
+                    if (((n_mode == 1) && ((socket_id < 0) || (socket_id == one_proc_table[index][PROC_SOCKET_ID]))) ||
+                        ((n_mode == 2) && (current_socket_id != one_proc_table[index][PROC_SOCKET_ID]) &&
+                         ((socket_id < 0) || (socket_id == one_proc_table[index][PROC_SOCKET_ID]))) ||
+                        ((n_mode == 3) && (current_socket_id == one_proc_table[index][PROC_SOCKET_ID]) &&
+                         ((socket_id < 0) || (socket_id == one_proc_table[index][PROC_SOCKET_ID])))) {
+                        if ((0 != one_proc_table[index][n]) && ((ALL_PROC == target_proc) || (n == target_proc))) {
```

Comment:
If customer use numactl, and `proc_type_table` is filtered by `numactl` which means that socket id or numa node id all ways starts from 0. But I think `current_socket_id` is real socket id. So I worried if there's something wrong with the value comparison here.

Reply:
`current_socket_id` is not the real socket id but socket id in _cpu_mapping_table(). If user us `numactl`, the id in _cpu_mapping_table() will be updated as well.